server/catgenome/src/main/java/com/epam/catgenome/manager/externaldb/homolog/HomologManager.java

```diff
@@ -0,0 +1,332 @@
+                    if (!CollectionUtils.isEmpty(geneAliases)) {
+                        List<String> aliasesList = geneAliases
+                                .stream()
+                                .map(Alias::getName)
+                                .collect(Collectors.toList());
+                        geneDesc.setAliases(aliasesList);
+                    }
+                    List<Domain> geneDomains = domains
+                            .stream()
+                            .filter(d -> d.getGeneId().equals(geneDesc.getGeneId()))
+                            .collect(Collectors.toList());
+                    if (!CollectionUtils.isEmpty(geneDomains)) {
+                        geneDesc.setDomains(geneDomains);
+                    }
+                    geneDescList.add(geneDesc);
+                } else {
+                    geneDescList.add(Gene.builder()
+                            .geneId(groupGene.getGeneId())
+                            .taxId(groupGene.getTaxId())
+                            .build());
+                }
+            }
+            group.setHomologs(geneDescList);
+        }
+    }
+
+    private QueryParameters buildSearchQuery(final HomologSearchRequest request) {
+        Assert.isTrue(!TextUtils.isBlank(request.getGeneId()), "Gene id is required");
+        final int pageNum = (request.getPage() == null || request.getPage() <= 0) ? 1 : request.getPage();
+        final int pageSize = (request.getPageSize() == null || request.getPage() <= 0) ? DEFAULT_PAGE_SIZE
+                : request.getPageSize();
+        PagingInfo pagingInfo = new PagingInfo(pageSize, pageNum);
+        QueryParameters queryParameters = QueryParameters.builder()
+                .pagingInfo(pagingInfo)
+                .build();
+        queryParameters.setFilters(Collections
+                .singletonList(new Filter("primary_gene_id", "=", request.getGeneId())));
+        return queryParameters;
+    }
+
+    @Transactional(propagation = Propagation.REQUIRED)
+    public void importHomologData(final String databaseName, final String databasePath)
+            throws IOException {
+        Assert.isTrue(!TextUtils.isBlank(databaseName), "Database name is required");
+        File file = new File(databasePath);
+        Assert.isTrue(file.isFile() && file.canRead(), getMessage(MessageCode.RESOURCE_NOT_FOUND));
+        deleteData(databaseName);
+        Map<Object, List<HomologRecord>> records = readLines(databasePath);
+        if (!CollectionUtils.isEmpty(records)) {
+            saveData(records, databaseName, databasePath);
+        }
+    }
+
+    public Map<Object, List<HomologRecord>> readLines(final String path) throws IOException {
+        List<HomologRecord> records = new ArrayList<>();
+        BufferedReader reader = new BufferedReader(new FileReader(path));
+        reader.readLine();
+        String line;
+        while ((line = reader.readLine()) != null) {
```

Comment:
I assume that ortholog files is organized by groups, so we can read group by group and save it to database chunk be chunk (save `N` groups at once), instead of loading everything in memory.